packages/foam-vscode/src/features/dataviz.ts

```diff
@@ -0,0 +1,111 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import { FoamFeature } from "../types";
+import { Foam } from "foam-core";
+import { TextDecoder } from "util";
+
+const feature: FoamFeature = {
+  activate: (context: vscode.ExtensionContext, foamPromise: Promise<Foam>) => {
+    vscode.commands.registerCommand("foam-vscode.show-graph", async () => {
+      const foam = await foamPromise;
+      const panel = vscode.window.createWebviewPanel(
+        "foam-graph",
+        "Foam Graph",
+        vscode.ViewColumn.Two,
+        {
+          enableScripts: true,
+          retainContextWhenHidden: true
+        }
+      );
+
+      const graph = {
+        nodes: [],
+        edges: []
+      };
+
+      foam.notes.getNotes().forEach(n => {
+        graph.nodes.push({
+          id: n.id,
+          type: "note",
+          uri: n.source.uri,
+          title: path.basename(n.source.uri),
+          nOutLinks: foam.notes.getForwardLinks(n.id).length,
+          nInLinks: foam.notes.getBacklinks(n.id).length
+        });
+        foam.notes.getForwardLinks(n.id).forEach(link => {
+          if (foam.notes.getNote(link.to) == null) {
```

Comment:
Is there a reason for the == over ===?

Reply:
`==` will compare both against `null` and `undefined`, whereas `===` would only do `null` (and we'd need another check for the `undefined` case).
I don't love it (it's the only place where I don't use the triple equal), but I like even less having both checks.

As I was writing I realized we have a `isNotNull` fn, we could create something along the same lines. 
To avoid loaded names, maybe we can use `isSome` / `isNone` (inspired by FP)